torch/csrc/jit/passes/utils/subgraph_utils.cpp

```diff
@@ -71,6 +71,7 @@ void mergeNodeIntoSubgraph(Node* toMerge, Node* subgraphNode) {
       // enable more optimizations
       if (auto value = toIValue(input)) {
         auto nv = subgraph->insertConstant(*value);
+        nv->setType(input->type()); // Need to retain type information on Nones
```

Comment:
Again, it didn't need to do it before, so why does it need to do it now? This is suspicious.

Reply:
Without setting the type toIValue was creating a None and inserting it. The type was going from t? to None, and then would fail at runtime in schema matching. 

Previously this wouldn't happen because toIValue didn't work on prim::None.

Reply:
I see. Right now the API is prone to mistakes, because it is easy to forget to call setType for the none case.  To fix this, I think `insertConstant` should gain a `TypePtr result_Type = nullptr` argument that, if present, is used as the output type with an assert that the that the IValue is compatible with that output type.